src/main/java/software/amazon/encryption/s3/materials/KmsKeyring.java

```diff
@@ -78,6 +78,11 @@ public String keyProviderInfo() {
             return KEY_PROVIDER_INFO;
         }
 
+        @Override
+        public boolean isKms() {
```

Comment:
This works, but it is not the best way to design the code. 
S3Keyring is an abstract class, so it should not contain any logic related to its implementing classes. 
It is better design if the S3Keyring stays abstract, though we do need to change some things to get this to work. 
We can also better adhere to the [MPL spec](https://github.com/awslabs/aws-encryption-sdk-specification/blob/abdd9483f9d2294ad1bc9ecda38ed61ef399b869/framework/keyring-interface.md). 
We should factor out `generateDataKey()` to be a method of the S3Keyring and its implementing classes. 
We'll need to move to passing `EncryptionMaterials` around though, since it has to add both the plaintext and encrypted data keys. 
Then in the S3Keyring we can follow the spec and only call the existing `encryptDataKey` method when the encrypted data key isn't already set.